Add Access-Control-Allow-Headers=Streamr-Client to CORS responses.

timoxley opened the pull request:
For CORE-1885, there's a new custom HTTP header sent along with client requests containing the streamr-client version:

![image](https://user-images.githubusercontent.com/43438/78160950-b3067680-7412-11ea-836c-6f31835eb5b7.png)

This header needs to be whitelisted otherwise the browser will prevent CORS requests that contain this header:

![image](https://user-images.githubusercontent.com/43438/78161204-1abcc180-7413-11ea-8c72-b3c06257279d.png)

Alternatively, this PR could be changed to allow all headers in CORS requests with `'Access-Control-Allow-Headers': '*'`

Also possible this should be changed in [`CorsResourceMapper.groovy`](https://github.com/streamr-dev/engine-and-editor/blob/master/grails-app/resourceMappers/CorsResourceMapper.groovy) instead.

timoxley commented:
Test failure seems unrelated.

kare commented:
@timoxley There was some availability issues with Maven repositories yesterday. Seems to work today. You probably want to merge master since I've just updated it with some REST API test timeout and Travis Node version fixes.

I think you can just test it by adding a new REST API test like here: https://github.com/streamr-dev/engine-and-editor/tree/master/rest-e2e-tests

timoxley commented:
Rebased on master, added a loose test for checking CORS headers https://github.com/streamr-dev/engine-and-editor/pull/746/commits/dca4d1136f57e08d51742f1f259393e69547e89e
